### DISCUSSION TITLE
Generating DAGs in epochs 472 and 473

### DISCUSSION THREAD

**Discussion by konstiantyn** [coinfoundry/miningcore#1158] [Q&A]
![image](https://user-images.githubusercontent.com/33037366/152833884-fc8fa11b-498a-4f3c-a544-ea92f46b0d1a.png)

I'm not sure what is problem. I really don't know if the pool works correctly or not. Pool hashrate dropped from 40GH/s to 0.

**Comment by konstiantyn** [comment]
![image](https://user-images.githubusercontent.com/33037366/152836659-e2ce9023-587b-49c3-ab76-d844735e9368.png)

Again generating the DAG in epoch 472.

**Reply by TooPlain** [reply]
Would suggest pre generating future dags epochs to avoid this. Should be in the geth manual.

**Reply by oliverw** [reply]
Any pointers where this is documented in the manual?

**Reply by MiningCryptoLive** [reply]
I am going to test this tonight but thought I would throw this out there.

According the Geth command line options the node when synced keeps 2 dag's on disc. default location is ~/.ethash
you also have the option to add dag's in mem using --ethash.dagsinmem (value) default is 1.

So we should be able to set the config file dag directory to ~/.ethash and miningcore should pick up the dag from the nodes dag directory instead of generating it. Correct me if I am wrong but I think this should work.

**Reply by oliverw** [reply]
That won't work currently but its an interesting idea to implement.

**Reply by MiningCryptoLive** [reply]
It would definitely help. Right now when a new dag comes we have to stop mining on the pool until the new dag generates. Which is sometimes close to an hour.

**Reply by oliverw** [reply]
Could you check if the latest `dev` branch improves dag generation times?

**Reply by MiningCryptoLive** [reply]
DAG Generation is still as slow on dev-branch as it is on the release branch.
